ocaml/xapi/xapi_network.ml

```diff
@@ -166,27 +183,46 @@ let pool_introduce ~__context ~name_label ~name_description ~mTU ~other_config ~
   let r = Ref.make() and uuid = Uuid.make_uuid() in
   Db.Network.create ~__context ~ref:r ~uuid:(Uuid.to_string uuid)
     ~current_operations:[] ~allowed_operations:[]
-    ~name_label ~name_description ~mTU ~bridge
+    ~name_label ~name_description ~mTU ~bridge ~managed:true
     ~other_config ~blobs:[] ~tags:[] ~default_locking_mode:`unlocked ~assigned_ips:[];
   r
 
-let create ~__context ~name_label ~name_description ~mTU ~other_config ~tags =
+let create ~__context ~name_label ~name_description ~mTU ~other_config ~bridge ~managed ~tags =
   Mutex.execute mutex (fun () ->
       let networks = Db.Network.get_all ~__context in
       let bridges = List.map (fun self -> Db.Network.get_bridge ~__context ~self) networks in
       let mTU = if mTU <= 0L then 1500L else mTU in
-      let rec loop () =
-        let name = stem ^ (string_of_int !counter) in
-        incr counter;
-        if List.mem name bridges then loop ()
-        else
-          let r = Ref.make () and uuid = Uuid.make_uuid () in
-          Db.Network.create ~__context ~ref:r ~uuid:(Uuid.to_string uuid)
-            ~current_operations:[] ~allowed_operations:[]
-            ~name_label ~name_description ~mTU ~bridge:name
-            ~other_config ~blobs:[] ~tags ~default_locking_mode:`unlocked ~assigned_ips:[];
-          r in
-      loop ())
+      let dbg = Context.string_of_task __context in
+      let ovs_bridges = Net.Bridge.get_all dbg () in
```

Comment:
Don't need this. It's fine if the bridge does not yet exist. By the way, in a pool you would need to check whether the bridge exists on all hosts, not just this one (which happens to be the pool master). So let's not do that.